src/main/java/com/mongodb/kafka/connect/sink/writemodel/strategy/DeleteOneBusinessKeyStrategy.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.kafka.connect.sink.writemodel.strategy;
+
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.ID_FIELD;
+import static com.mongodb.kafka.connect.sink.writemodel.strategy.WriteModelHelper.flattenKeys;
+
+import org.apache.kafka.connect.errors.DataException;
+
+import org.bson.BSONException;
+import org.bson.BsonDocument;
+
+import com.mongodb.client.model.DeleteOneModel;
+import com.mongodb.client.model.WriteModel;
+
+import com.mongodb.kafka.connect.sink.Configurable;
+import com.mongodb.kafka.connect.sink.MongoSinkTopicConfig;
+import com.mongodb.kafka.connect.sink.converter.SinkDocument;
+import com.mongodb.kafka.connect.sink.processor.id.strategy.IdStrategy;
+import com.mongodb.kafka.connect.sink.processor.id.strategy.PartialKeyStrategy;
+import com.mongodb.kafka.connect.sink.processor.id.strategy.PartialValueStrategy;
+
+public class DeleteOneBusinessKeyStrategy implements WriteModelStrategy, Configurable {
+
+  private boolean isPartialId = false;
+
+  @Override
+  public WriteModel<BsonDocument> createWriteModel(final SinkDocument document) {
+    BsonDocument vd =
+        document
+            .getValueDoc()
+            .orElseThrow(
+                () ->
+                    new DataException(
+                        "Could not build the WriteModel,the value document was missing unexpectedly"));
+
+    try {
+      BsonDocument businessKey = vd.getDocument(ID_FIELD);
+      vd.remove(ID_FIELD);
+      if (isPartialId) {
+        businessKey = flattenKeys(businessKey);
+      }
+      return new DeleteOneModel<>(businessKey);
+
+    } catch (BSONException e) {
+      throw new DataException(
```

Comment:
It appears that there are too many statements between `vd.getDocument(ID_FIELD)` and `catch (BSONException e)` preventing us to state that any `BSONException` caught was thrown by the `vd.getDocument(ID_FIELD)` statement.

I propose to shrink the `try` block so that it contains only `vd.getDocument(ID_FIELD)`.

Reply:
Agreed - will do as a separate commit as this is a pattern is repeated in all the BusinessKey based write strategies

Reply:
Done - I've removed the try catch and bought it inline with the other checks idValue checks added.